src/exec.js

```diff
@@ -104,8 +105,11 @@ function execSync(cmd, opts, pipe) {
     code = parseInt(fs.readFileSync(codeFile, 'utf8'), 10);
   }
 
-  var stdout = fs.readFileSync(stdoutFile, 'utf8');
-  var stderr = fs.readFileSync(stderrFile, 'utf8');
+  // fs.readFileSync uses buffer encoding by default, so set
+  // the encoding only if opts.encoding wasn't set to 'buffer'
+  var encoding = opts.encoding !== 'buffer' ? opts.encoding : null;
```

Comment:
Perhaps this would be clearer:
```
var stdout;
var stderr;
if (opts.encoding === 'buffer') {
    stdout = fs.readFileSync(stdoutFile);
    stderr = fs.readFileSync(stderrFile);
} else {
    stdout = fs.readFileSync(stdoutFile, opts.encoding);
    stderr = fs.readFileSync(stderrFile, opts.encoding);
}
```